codex-rs/core/src/codex.rs

```diff
@@ -832,6 +935,15 @@ async fn run_task(sess: Arc<Session>, sub_id: String, input: Vec<InputItem>) {
                     .collect::<Vec<ResponseInputItem>>();
                 let last_assistant_message = get_last_assistant_message_from_turn(&items);
 
+                // When usage info not provided (usage_out==0) fall back to approximate.
+                if usage_out == 0 {
+                    total_completion_tokens += count_tokens_in_items(&items);
+                }
```

Comment:
This seems odd - if the backing api or model does not support accurate usage I'd rather not return or display token usage whatsoever. It's better to display nothing than to not be accurate re costs.

Is there a better way to determine if it does?